app/components/radio-selector.js

```diff
@@ -0,0 +1,58 @@
+import Ember from 'ember';
+import { set } from '@ember/object';
+import { copy } from '@ember/object/internals';
+import { merge } from '@ember/polyfills';
+import { next } from '@ember/runloop';
+
+const hideLayer = (layer) => {
+  const layerCopy = copy(layer, true);
+  layerCopy.layer.layout = merge(layerCopy.layer.layout, { visibility: 'none' });
+  set(layer, 'layer', layerCopy.layer);
+};
+
+const showLayer = (layer) => {
+  const layerCopy = copy(layer, true);
+  layerCopy.layer.layout = merge(layerCopy.layer.layout, { visibility: 'visible' });
+  set(layer, 'layer', layerCopy.layer);
+};
+
+
+export default Ember.Component.extend({
+  init(...args) {
+    this._super(...args);
+
+    next(() => {
+      const layers = this.get('layers');
+      const qps = this.get('qps');
+
+      const matchedLayer = layers.find(layer => qps.get(layer.layer.id) === true);
+
+      if (matchedLayer) {
+        this.send('switchLayer', matchedLayer.layer.id);
+      }
+    });
+  },
```

Comment:
We talked a bit about this larger design issue - overriding init and conditionally determining an initial state is something we do elsewhere, but I wonder if it's a code smell? Basically, we have query params, which have defaults that are determined by configs, but query params are allowed to be set by the URL. Basically, I wonder if this component can just be responsible for managing the layer visibility state, and this overriding behavior happens somewhere else higher up in the app.